RosettaCore/src/main/java/com/hubspot/rosetta/internal/RosettaAnnotationIntrospector.java

```diff
@@ -211,4 +217,13 @@ private Annotated getAnnotatedTypeFromAnnotatedMethod(AnnotatedMethod a) {
       );
     }
   }
+
+  private <T> Optional<T> getFirstNonEmpty(Supplier<Optional<T>>... suppliers) {
+    return Arrays
+      .stream(suppliers)
+      .map(Supplier::get)
+      .filter(Optional::isPresent)
```

Comment:
I think this should be fine in practice, but if the implementation is ungenerous we'll end up calling `get` needlessly. I think a for-loop might be best for this one.